test/humanStandardToken.js

```diff
@@ -1,345 +1,204 @@
-      ctr = result
-      return ctr.approve(accounts[1], 100, {from: accounts[0]})
-    }).then(function (result) {
-      return ctr.allowance.call(accounts[0], accounts[1])
-    }).then(function (result) {
-      assert.strictEqual(result.toNumber(), 100)
-      return ctr.transferFrom(accounts[0], accounts[2], 50, {from: accounts[1]})
-    }).then(function (result) {
-      return ctr.allowance.call(accounts[0], accounts[1])
-    }).then(function (result) {
-      assert.strictEqual(result.toNumber(), 50)
-      return ctr.balanceOf.call(accounts[2])
-    }).then(function (result) {
-      assert.strictEqual(result.toNumber(), 50)
-      return ctr.balanceOf.call(accounts[0])
-    }).then(function (result) {
-      assert.strictEqual(result.toNumber(), 9950)
-            // FIRST tx done.
-            // onto next.
-      return ctr.transferFrom.call(accounts[0], accounts[2], 60, {from: accounts[1]})
-    }).then(function (result) {
-      assert(false, 'The preceding call should have thrown an error.')
-    }).catch((err) => {
-      assert(evmThrewError(err), 'the EVM did not throw an error or did not ' +
-                                 'throw the expected error')
-    })
+  // should approve 100 of msg.sender & withdraw 50 & 60 (should fail).
+  it('approvals: msg.sender approves accounts[1] of 100 & withdraws 50 & 60 (2nd tx should fail)', async () => {
+    await HST.approve(accounts[1], 100, {from: accounts[0]})
+    const allowance01 = await HST.allowance.call(accounts[0], accounts[1])
+    assert.strictEqual(allowance01.toNumber(), 100)
+
+    await HST.transferFrom(accounts[0], accounts[2], 50, {from: accounts[1]})
+    const allowance012 = await HST.allowance.call(accounts[0], accounts[1])
+    assert.strictEqual(allowance012.toNumber(), 50)
+
+    const balance2 = await HST.balanceOf.call(accounts[2])
+    assert.strictEqual(balance2.toNumber(), 50)
+
+    const balance0 = await HST.balanceOf.call(accounts[0])
+    assert.strictEqual(balance0.toNumber(), 9950)
+
+    // FIRST tx done.
+    // onto next.
+    await expectThrow(HST.transferFrom.call(accounts[0], accounts[2], 60, {from: accounts[1]}))
   })
 
-  it('approvals: attempt withdrawal from acconut with no allowance (should fail)', function () {
-    var ctr = null
-    return HumanStandardToken.new(10000, 'Simon Bucks', 1, 'SBX', {from: accounts[0]}).then(function (result) {
-      ctr = result
-      return ctr.transferFrom.call(accounts[0], accounts[2], 60, {from: accounts[1]})
-    }).then(function (result) {
-      assert(false, 'The preceding call should have thrown an error.')
-    }).catch((err) => {
-      assert(evmThrewError(err), 'the EVM did not throw an error or did not ' +
-                                 'throw the expected error')
-    })
+  it('approvals: attempt withdrawal from account with no allowance (should fail)', function () {
+    return expectThrow(HST.transferFrom.call(accounts[0], accounts[2], 60, {from: accounts[1]}))
```

Comment:
@Janaka-Steph I noticed that sometimes you use `return expectThrow(...`, and others (ie. line 172 below), you use `await expectThrow(...`. 

Any reason for the difference? It's a bit odd that both work.

Reply:
Good question. Mocha test is waiting for a promise. But async/await does some magic and if you don't explicitly return a promise, the value you return will automatically be wrapped in a promise. So I think it doesn't matter here if we write `await` or `return`. But not writing neither of them should lead to unexpected issues.
This thread can be interesting to read: https://stackoverflow.com/questions/35302431/async-await-implicitly-returns-promise

Reply:
Good article, thank you. 

Do you think using `return` would make more sense semantically, as `expectThrow` is not returning a promise?

Reply:
Yes you are right, it would be better with a `return`.